projects/packages/stats-admin/src/class-rest-controller.php

```diff
@@ -41,33 +58,41 @@ public function register_rest_routes() {
 				'permission_callback' => array( $this, 'check_user_privileges_callback' ),
 			)
 		);
+
+		// Stats for a resource type.
 		register_rest_route(
 			static::$namespace,
 			sprintf( '/sites/%d/stats/(?P<resource>[\-\w]+)', Jetpack_Options::get_option( 'id' ) ),
 			array(
 				'methods'             => WP_REST_Server::READABLE,
-				'callback'            => array( $this, 'get_stats_resource_from_wpcom' ),
+				'callback'            => array( $this, 'get_stats_resource' ),
 				'permission_callback' => array( $this, 'check_user_privileges_callback' ),
 			)
 		);
+
+		// Single post info.
 		register_rest_route(
 			static::$namespace,
-			sprintf( '/sites/%d/(?P<resource>[\-\w]+)/(?P<resource_id>[\d]+)/(?P<sub_resource>[\-\w]+)', Jetpack_Options::get_option( 'id' ) ),
+			sprintf( '/sites/%d/posts/(?P<resource_id>[\d]+)', Jetpack_Options::get_option( 'id' ) ),
 			array(
 				'methods'             => WP_REST_Server::READABLE,
-				'callback'            => array( $this, 'get_site_sub_resource_from_wpcom' ),
+				'callback'            => array( $this, 'get_single_post' ),
 				'permission_callback' => array( $this, 'check_user_privileges_callback' ),
 			)
 		);
+
+		// Single post likes.
 		register_rest_route(
 			static::$namespace,
-			sprintf( '/sites/%d/(?P<resource>[\-\w]+)/(?P<resource_id>[\d]+)', Jetpack_Options::get_option( 'id' ) ),
+			sprintf( '/sites/%d/posts/(?P<resource_id>[\d]+)/likes', Jetpack_Options::get_option( 'id' ) ),
```

Comment:
This will not work without the Jetpack plugin present. Also forwarding a request to WPCOM, only to be forwarded back to the site doesn't make much sense. I'll try to think of an alternative solution for this case.

Reply:
And we'll need to allow blog auth for the endpoint if I'm not getting it wrong.

Reply:
Update:

Seems the endpoint is public 😄

Reply:
changed the API to redirection atm d7c3b5b949

Will look at the possibility to access directly from the front end if there are other public endpoints access in the next stage.